services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -3720,19 +3720,22 @@ Pair<Store, Version> waitVersion(String clusterName, String storeName, int versi
   @Override
   public void setStoreCurrentVersion(String clusterName, String storeName, int versionNumber) {
     storeMetadataUpdate(clusterName, storeName, store -> {
-      if (store.getCurrentVersion() != Store.NON_EXISTING_VERSION) {
-        if (versionNumber != Store.NON_EXISTING_VERSION && !store.containsVersion(versionNumber)) {
-          throw new VeniceException("Version:" + versionNumber + " does not exist for store:" + storeName);
-        }
+      // Parent colo should not update the current version of a store
+      if (!isParent()) {
+        if (store.getCurrentVersion() != Store.NON_EXISTING_VERSION) {
+          if (versionNumber != Store.NON_EXISTING_VERSION && !store.containsVersion(versionNumber)) {
+            throw new VeniceException("Version:" + versionNumber + " does not exist for store:" + storeName);
+          }
 
-        if (!store.isEnableWrites()) {
-          throw new VeniceException(
-              "Unable to update store:" + storeName + " current version since store writeability is false");
+          if (!store.isEnableWrites()) {
+            throw new VeniceException(
+                "Unable to update store:" + storeName + " current version since store writeability is false");
```

Comment:
nit: (that I think 'I' originally did, but I'll make you fix it XD) add a space after store: and Version: in the two exception messages.